src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -4820,77 +4820,77 @@ public String getResolutionForKeepAR(int scaleWidth, int scaleHeight) {
-			}
+		// If the in-memory media has not already been populated with filename metadata, we attempt it
+		try {
+			if (isBlank(media.getMovieOrShowName())) {
+				String[] metadataFromFilename = FileUtil.getFileNameMetadata(file.getName(), file.getAbsolutePath());
+				String titleFromFilename = metadataFromFilename[0];
+				String yearFromFilename = metadataFromFilename[1];
+				String extraInformationFromFilename = metadataFromFilename[2];
+				String tvSeasonFromFilename = metadataFromFilename[3];
+				String tvEpisodeNumberFromFilename = metadataFromFilename[4];
+				String tvEpisodeNameFromFilename = metadataFromFilename[5];
+				String titleFromFilenameSimplified = FileUtil.getSimplifiedShowName(titleFromFilename);
+
+				media.setMovieOrShowName(titleFromFilename);
+				media.setSimplifiedMovieOrShowName(titleFromFilenameSimplified);
+				String titleFromDatabase;
+				String titleFromDatabaseSimplified;
+
+				// Apply the metadata from the filename.
+				if (isNotBlank(titleFromFilename) && isNotBlank(tvSeasonFromFilename)) {
+					/**
+					* Overwrite the title from the filename if it's very similar to one
+					* we already have in our database. This is to avoid minor
+					* grammatical differences like "Word and Word" vs. "Word & Word"
+					* from creating two virtual folders.
+					*/
+					titleFromDatabase = TableTVSeries.getSimilarTVSeriesName(titleFromFilename);
+					titleFromDatabaseSimplified = FileUtil.getSimplifiedShowName(titleFromDatabase);
+					if (titleFromFilenameSimplified.equals(titleFromDatabaseSimplified)) {
+						media.setMovieOrShowName(titleFromDatabase);
+					}
 
-			media.setTVSeason(tvSeasonFromFilename);
-			if (isNotBlank(tvEpisodeNumberFromFilename)) {
-				media.setTVEpisodeNumber(tvEpisodeNumberFromFilename);
-			}
-			if (isNotBlank(tvEpisodeNameFromFilename)) {
-				media.setTVEpisodeName(tvEpisodeNameFromFilename);
-			}
+					media.setTVSeason(tvSeasonFromFilename);
+					if (isNotBlank(tvEpisodeNumberFromFilename)) {
+						media.setTVEpisodeNumber(tvEpisodeNumberFromFilename);
+					}
+					if (isNotBlank(tvEpisodeNameFromFilename)) {
+						media.setTVEpisodeName(tvEpisodeNameFromFilename);
+					}
 
-			media.setIsTVEpisode(true);
-		}
+					media.setIsTVEpisode(true);
+				}
 
-		if (yearFromFilename != null) {
-			media.setYear(yearFromFilename);
-		}
+				if (yearFromFilename != null) {
+					media.setYear(yearFromFilename);
+				}
 
-		if (extraInformationFromFilename != null) {
-			media.setExtraInformation(extraInformationFromFilename);
-		}
+				if (extraInformationFromFilename != null) {
+					media.setExtraInformation(extraInformationFromFilename);
+				}
 
-		try {
-			if (configuration.getUseCache()) {
-				PMS.get().getDatabase().insertVideoMetadata(file.getAbsolutePath(), file.lastModified(), media);
+				if (configuration.getUseCache()) {
+					// TODO: Make sure this does not happen if ANY version already exists, before doing this
```

Comment:
Just to double check - these are todos not to be addressed in this PR?

Reply:
Good question, yes I think it can wait since that's existing code